src/invidious/routes/watch.cr

```diff
@@ -275,6 +275,12 @@ module Invidious::Routes::Watch
     return error_template(400, "Invalid clip ID") if response["error"]?
 
     if video_id = response.dig?("endpoint", "watchEndpoint", "videoId")
+      if params = response.dig?("endpoint", "watchEndpoint", "params").try &.as_s
+        start_time, end_time, _ = parse_clip_parameters(params)
+        env.params.query["start"] = start_time.to_s
+        env.params.query["end"] = end_time.to_s
```

Comment:
Depending on how you'll handle an unexpected parsing error at `parse_clip_parameters`, those might need some `nil` checks